### ISSUE TITLE
About Rigidity masks 

### ISSUE THREAD

**Issue by ezorfa** [anuragranj/cc#18]
Hi @anuragranj !

I am training the model using all the pre-trained models that you published online. However, I do not get proper rigidity masks. Do you know why this is happening?  As I understand, Rigidity masks are the subtraction of rigid flow with flow generated by the flownet with some threshold(I havent chaned the code at all). Below is the picture:

<img width="1086" alt="Screenshot 2019-11-26 at 07 21 07" src="https://user-images.githubusercontent.com/50787308/69604463-d745e700-101d-11ea-82f4-495a724bf2e5.png">

Also, Could you share the optimizer checkpoint at the time of your training, so that I could visualize all the results at once?

**Comment by anuragranj** [comment]
Can you share the hyperparameters you are using for training? Also, are the depth, camera and flow networks trained well?

**Comment by ezorfa** [comment]
Thankyou @anuragranj ! Unfortunately, I cannot share with you immediately as I am not at my desk. However, I will do it as soon. If possible, meanwhile, could you share your tensorboard events file wherein I can see the rigid masks, please.

> Also, are the depth, camera and flow networks trained well?

I have plugged in your pretrained depth, camera, and flow models. So, Isnt it already trained?

**Comment by anuragranj** [comment]
I don't have access to tensorboard event files. Could  you try the following hyperparameters? This worked for me.

```sh
python3 train.py DATA_PATH --sequence-length 5 --dispnet DispResNet6 \
--posenet PoseNetB6 --masknet MaskNet6 --flownet Back2Future --pretrained-disp \
pretrained/dispnet_model_best.pth.tar --pretrained-pose pretrained/posenet_model_best.pth.tar \
--pretrained-mask pretrained/masknet_model_best.pth.tar --pretrained-flow \
pretrained/flownet_model_best.pth.tar -b4 -m0.02 -pf 0.5 -pc 1.0 -s0.2 -c0.3 \
--epoch-size 100 --log-output -f 0 --nlevels 6 --lr 1e-6 -wssim 0.5 --with-flow-gt \
--with-depth-gt --epochs 100 --fix-flownet --fix-dispnet --fix-posenet \
--smoothness-type edgeaware --THRESH 0.999
```

**Comment by ezorfa** [comment]
Hi @anuragranj !

I used the default THRESH value of 0.01 before. 

However, I ran `test_flow.py` script with the THRESH value of 0.999. I expect the rigidity mask to be as shown in the paper:

<img width="339" alt="Screenshot 2019-11-26 at 20 30 49" src="https://user-images.githubusercontent.com/50787308/69665932-a8665a00-108b-11ea-890d-c498928613b0.png">


However, while running the command:

`python test_flow.py  --pretrained-disp /cdtemp/ezorfa/cc-master/checkpoints/dispnet_k.pth.tar  --pretrained-pose /cdtemp/ezorfa/cc-master/checkpoints/posenet.pth.tar --pretrained-flow /cdtemp/ezorfa/cc-master/checkpoints/back2future.pth.tar   --pretrained-mask /cdtemp/ezorfa/cc-master/checkpoints/masknet.pth.tar --THRESH 0.999 --kitti-dir kitti2015/dataset --output-dir kitti2015/output/`  I get outputs of the rigidity mask like: 

<img width="557" alt="Screenshot 2019-11-26 at 20 28 26" src="https://user-images.githubusercontent.com/50787308/69665749-51608500-108b-11ea-9c6c-b2a8e6336f62.png">

**How can I get the mask outputs like the one described in the paper?** 

Thankyou!

**Comment by ezorfa** [comment]
@anuragranj  Is such a mask expected when camera motion is present? Also, could you please check the parameter values used in the above command `test_flow.py` and tell me how to reproduce rigidity_flow_census masks as shown in the paper.  

I am only interested to find the moving objects,  by making use of rigidity flow and optical flow obtained by flownet and subsequently thresholding.

Thankyou!

**Comment by anuragranj** [comment]
To get the masks, you should be running test_mask.py with default  --THRESH values. The test_flow.py functions outputs the masks that are optimal for best flow, but not for best masks.

**Comment by ezorfa** [comment]
@anuragranj  Okay, Could you please assert if such masks like the picture above are expected?

**Comment by anuragranj** [comment]
Masks like Figure 1. in the paper are expected if you use test_mask.py with default --THRESH, which is .94.

**Comment by ezorfa** [comment]
Hmm. I am testing now. Thankyou for the guidance.

**Comment by anuragranj** [comment]
The visualization corresponds to the following line:
https://github.com/anuragranj/cc/blob/master/test_mask.py#L173
It is saved as 'val Rigidity Mask Census' in tensorboard events.

**Comment by ezorfa** [comment]
yeah, I spotted that and made my own script to vizualize only moving objects but somehow dint see the file test_mask.py existing. Sorry about that and will come back to you.
